src/gluonts/dataset/pandas.py

```diff
@@ -11,39 +11,30 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-from copy import deepcopy
+from typing import Any, Dict, Iterable, Union, List, Optional, Tuple
 from dataclasses import dataclass, field
-from typing import (
-    Any,
-    Collection,
-    Dict,
-    Iterator,
-    Iterable,
-    List,
-    Optional,
-    Union,
-    Tuple,
-)
 
-import pandas as pd
 import numpy as np
+import pandas as pd
+from pandas.api.types import is_numeric_dtype
 from pandas.core.indexes.datetimelike import DatetimeIndexOpsMixin
 from toolz import first
 
-from gluonts.dataset.common import DataEntry, _as_period
-from gluonts.dataset.field_names import FieldName
-from gluonts.itertools import Map, SizedIterable
+from gluonts.dataset.common import DataEntry
+from gluonts.itertools import SizedIterable, Map
 
 
 @dataclass
 class PandasDataset:
     """
-    A pandas.DataFrame-based dataset type.
+    A dataset type based on ``pandas.DataFrame``.
 
-    This class is constructed with a collection of pandas.DataFrame-objects
-    where each ``DataFrame`` is representing one time series.
-    A ``target`` and a ``timestamp`` columns are essential. Furthermore,
-    static/dynamic real/categorical features can be specified.
+    This class is constructed with a collection of ``pandas.DataFrame``
+    objects where each ``DataFrame`` is representing one time series.
+    A ``target`` and a ``timestamp`` columns are essential. Dynamic features
+    of a series can be specified with together with the series' ``DataFrame``,
+    while static features can be specified in a separate ``DataFrame`` object
```

Comment:
Can I have static features in a column?

Reply:
No

Reply:
To expand a bit on this -- of course, one could have static features as columns to begin with: when that's the case, they can be extracted into a separate data frame pretty easily. For example 

```python
import pandas as pd

df = pd.DataFrame(
    data={
        'item_id': [1, 1, 2, 2],
        'static_feature': ['a', 'a', 'b', 'b'],
        'target': [1.0, 2.0, 3.0, 4.0]
    }
)

df[["item_id", "static_feature"]].drop_duplicates().set_index("item_id")
```

gives 

```
        static_feature
item_id
1                    a
2                    b
```

On the other hand, forcing one to specify static features as columns is inefficient when datasets are big (see M5 example above). Also, I think that offering this single entry point for static features is better since it allows us to do encoding automatically, while this would be difficult in case each input dataframe had its own columns of static features (need to go through the whole dataset and find all values and so on).

Reply:
I guess the above ☝️ could be automated in `from_long_dataframe` actually

Reply:
(Done)